subprojects/docs/docs.gradle

```diff
@@ -447,17 +280,109 @@ tasks.register("checkstyleApi", Checkstyle) {
     reports.xml.destination = file("$checkstyle.reportsDir/checkstyle-api.xml")
 }
 
-def distDocs = tasks.register("distDocs", Docbook2Xhtml) {
-    dependsOn userguideAsciidoc
-    source new File(asciidocOutputDir, 'getting_started.xml')
-    destFile = new File(distDocsDir, 'getting-started.html')
-    stylesheetName = 'standaloneHtml.xsl'
+def distDocs = tasks.register("distDocs", AsciidoctorTask) {
+    sourceDir = userguideSrcDir
+    sources { include 'getting-started.adoc' }
+    outputDir = distDocsDir
+    separateOutputDirs = false
+    options doctype: 'book'
+    backends 'html5'
+}
+
+asciidoctorj {
+    noDefaultRepositories = true
+    version = '1.5.7'
 }
 
-def userguide = tasks.register("userguide") {
-    dependsOn userguideHtml, userguideSingleHtml, userguidePdf
-    description = 'Generates the userguide'
+def userguideAsciidoc = tasks.register("userguideAsciidoc", AsciidoctorTask) {
+    dependsOn css, samples, defaultImports
+    sourceDir = userguideSrcDir
+
+    inputs.file "$srcDocsDir/css/manual.css"
+    inputs.dir file("src/main/resources")
+    inputs.dir samplesSrcDir
+    sources {
+        include '*.adoc'
```

Comment:
IIUC, we are now processing each `.adoc` file twice, once individually and once via `userguide_single.adoc`. Is this correct?

The single processed `.adoc` files also seem to have broken links when referencing to another chapter via e.g. `<<sec:up_to_date_checks,incremental build>>`. This works in `userguide_single`. Will this be fixed in the final toolchain? How do you think to generate the individual html pages?

Reply:
We should exclude userguide_single here, and make the PDF task just be single-page HTML and PDF. 

In #5837 and other PRs I will submit, the links are all re-written to work with the single-page, multi-page, and PDF. That's why I mention in the description of #5837 that "Users must use a <file>.adoc# prefix for all cross-reference links."  Asciidoc just doesn't seem to have a more robust solution for this, and I really don't want to build a bunch more to support it.

Reply:
Got it.